### ISSUE TITLE
XL: Rename to tmp location before deleting the existing object. 

### ISSUE THREAD

**Issue by harshavardhana** [minio/minio#1697]
xl-objects-multipart.go

```
      // Delete if an object already exists.
        // FIXME: rename it to tmp file and delete only after
        // the newly uploaded file is renamed from tmp location to
        // the original location.
        err = xl.deleteObject(bucket, object)
        if err != nil && err != errFileNotFound {
                return "", toObjectErr(err, bucket, object)
        }
```

xl-objects.go

```
 // Delete if an object already exists.
        // FIXME: rename it to tmp file and delete only after
        // the newly uploaded file is renamed from tmp location to
        // the original location.
        err = xl.deleteObject(bucket, object)
        if err != nil && err != errFileNotFound {
                return "", toObjectErr(err, bucket, object)
        }
```

**Comment by balamurugana** [comment]
To be fixed by 20-May-2016

**Comment by harshavardhana** [comment]
Fixed by #1790
